lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -453,6 +460,68 @@ void SPIRVRegularizeLLVMBase::lowerFuncPtr(Module *M) {
     lowerFuncPtr(I.first, I.second);
 }
 
+void SPIRVRegularizeLLVMBase::addKernelEntryPoint(Module *M) {
+  std::vector<Function *> Work;
+
+  // Get a list of all functions that have SPIR kernel calling conv
```

Comment:
Should we process only called kernels? I believe that the kernels which are not called might not need this transformation.

Reply:
can we tell that, especially for linkages, if something is called from another module?

Reply:
I wasn't sure that OpenCL actually allows to call the kernels from other translation units. Reading comments to https://github.com/KhronosGroup/SPIRV-Registry/issues/74 , seems this might be the case. If someone really links on SPIR-V level (and I guess it is also possible) we should really wrap them all...